sdk/state/integrationtests/helpers_test.go

```diff
@@ -215,13 +225,15 @@ func randomPositiveInt64(t *testing.T, max int64) int64 {
 	return int64(i) % max
 }
 
-func retry(maxAttempts int, f func() error) (err error) {
+func retry(t *testing.T, maxAttempts int, f func() error) (err error) {
+	t.Helper()
 	for i := 0; i < maxAttempts; i++ {
 		err = f()
 		if err == nil {
 			return
 		}
-		time.Sleep(time.Second)
+		t.Logf("failed attempt %d at performing a retry-able operation: %v", i, err)
+		time.Sleep(5 * time.Second)
```

Comment:
❓ is a full 5 seconds needed? In thoery seems long to me and could slow down the workflows a noticeable amount. But if 5 is what you found is needed empirically then sgtm 👍

Reply:
Let's try 2.
```suggestion
		time.Sleep(2 * time.Second)
```